During build, I get this error:
```
Exception in thread "Isabelle.thread-11" Exception in thread "Isabelle.thread-15" java.io.IOException: Cannot run program "/usr/bin/env": error=2, No such file or directory
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
        at isabelle.setup.Environment.exec_process(Environment.java:234)
        at isabelle.setup.Environment.kill_process(Environment.java:451)
        at isabelle.Bash$Process.signal(bash.scala:130)
        at isabelle.Bash$Process.$anonfun$terminate$1(bash.scala:144)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
        at isabelle.Isabelle_Thread.interrupt_handler(isabelle_thread.scala:188)
        at isabelle.Isabelle_Thread$.interrupt_handler(isabelle_thread.scala:116)
        at isabelle.Isabelle_Thread$.try_uninterruptible(isabelle_thread.scala:128)
        at isabelle.Bash$Process.terminate(bash.scala:143)
        at isabelle.Bash$Process.$anonfun$shutdown_hook$1(bash.scala:157)
        at isabelle.Isabelle_Thread.run(isabelle_thread.scala:141)
Caused by: java.io.IOException: error=2, No such file or directory
        at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
        at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:314)
        at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:244)
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1110)
        ... 12 more
java.io.IOException: Cannot run program "/usr/bin/env": error=2, No such file or directory
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
        at isabelle.setup.Environment.exec_process(Environment.java:234)
        at isabelle.setup.Environment.kill_process(Environment.java:451)
        at isabelle.Bash$Process.signal(bash.scala:130)
        at isabelle.Bash$Process.$anonfun$terminate$1(bash.scala:144)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
        at isabelle.Isabelle_Thread.interrupt_handler(isabelle_thread.scala:188)
        at isabelle.Isabelle_Thread$.interrupt_handler(isabelle_thread.scala:116)
        at isabelle.Isabelle_Thread$.try_uninterruptible(isabelle_thread.scala:128)
        at isabelle.Bash$Process.terminate(bash.scala:143)
        at isabelle.Bash$Process.$anonfun$shutdown_hook$1(bash.scala:157)
        at isabelle.Isabelle_Thread.run(isabelle_thread.scala:141)
Caused by: java.io.IOException: error=2, No such file or directory
        at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
        at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:314)
        at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:244)
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1110)
        ... 12 more
```

The build continues though, and the prebuilt HOL session seems to work fine.  IIRC a long time ago we prebuilt the HOL session, but I've disabled it in 2016 because it didn't seem to work (the HOL session was still getting built on first startup).  It's great to see this working again!

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -5999,6 +5999,16 @@
     githubId = 2396926;
     name = "Justin Woo";
   };
+  jvanbruegge = {
+    email = "supermanitu@gmail.com";
+    github = "jvanbruegge";
+    githubId = 1529052;
+    name = "Jan van Brügge";
+    keys = [{
+      longkeyid = "rsa4096/0x366572BE7D6C78A2";
+      fingerprint = "3513 5CE5 77AD 711F 3825  9A99 3665 72BE 7D6C 78A2";
+    }];
+  };
   jwatt = {
     email = "jwatt@broken.watch";
     github = "jjwatt";

--- a/pkgs/applications/science/logic/isabelle/default.nix
+++ b/pkgs/applications/science/logic/isabelle/default.nix
@@ -127,7 +127,7 @@ stdenv.mkDerivation rec {
     '';
     homepage = "https://isabelle.in.tum.de/";
     license = licenses.bsd3;
-    maintainers = [ maintainers.jwiegley ];
+    maintainers = [ maintainers.jwiegley maintainers.jvanbruegge ];
     platforms = platforms.linux;
   };
 }

--- a/pkgs/applications/science/logic/isabelle/default.nix
+++ b/pkgs/applications/science/logic/isabelle/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchurl, nettools, java, polyml, z3, veriT, vampire, eprover-ho, rlwrap, makeDesktopItem }:
+{ lib, stdenv, fetchurl, coreutils, nettools, java, polyml, z3, veriT, vampire, eprover-ho, rlwrap, makeDesktopItem }:
 # nettools needed for hostname
 
 stdenv.mkDerivation rec {
@@ -73,6 +73,11 @@ stdenv.mkDerivation rec {
     for comp in contrib/jdk* contrib/polyml-* contrib/z3-* contrib/verit-* contrib/vampire-* contrib/e-*; do
       rm -rf $comp/x86*
     done
+
+    substituteInPlace lib/Tools/env \
+      --replace /usr/bin/env ${coreutils}/bin/env
+
+    rm -r heaps
   '' + (if ! stdenv.isLinux then "" else ''
     arch=${if stdenv.hostPlatform.system == "x86_64-linux" then "x86_64-linux" else "x86-linux"}
     for f in contrib/*/$arch/{bash_process,epclextract,nunchaku,SPASS,zipperposition}; do
@@ -82,6 +87,11 @@ stdenv.mkDerivation rec {
       patchelf --set-rpath "${lib.concatStringsSep ":" [ "${java}/lib/openjdk/lib/server" "${stdenv.cc.cc.lib}/lib" ]}" $d/*.so
     done
   '');
+
+  buildPhase = ''
+    export HOME=$TMP # The build fails if home is not set
+    bin/isabelle build -v -o system_heaps -b HOL
+  '';
 
   installPhase = ''
     mkdir -p $out/bin

